pulp_ansible/app/tasks/collections.py

```diff
@@ -699,21 +711,38 @@ async def _add_namespace(self, name):
             pass
         else:
             namespace = parse_metadata(result)
-            namespace.pop("pulp_href", None)
+            links = namespace.get("links", None)
+
+            # clean up the galaxy API for pulp
+            if links:
+                namespace["links"] = {x["name"]: x["url"] for x in links}
+            else:
+                namespace["links"] = dict()
+
+            for key in ("pulp_href", "groups", "id", "related_fields"):
+                namespace.pop(key, None)
+
+            avatar_sha256 = namespace.get("avatar_sha256", None)
             url = namespace.pop("avatar_url", None)
-            da = (
-                [
-                    DeclarativeArtifact(
-                        Artifact(sha256=namespace["avatar_sha256"]),
+
+            logo = None
+            # don't download the logo if it already exists
+            if not sync_to_async(Artifact.objects.filter(sha256=avatar_sha256).exists):
+                if url:
+                    logo = DeclarativeArtifact(
+                        Artifact(sha256=avatar_sha256),
                         url=url,
                         remote=self.remote,
                         relative_path=f"{name}-avatar",
                         deferred_download=False,
                     )
-                ]
-                if url
-                else None
-            )
+
+                    await logo.download()
+                else:
+                    logo = None
```

Comment:
This doesn't make sense. The pipeline already takes care of checking if the artifact exists based on the `avatar_sha256` and figuring out if it should be downloaded. Since you are not using the artifact immediately now, revert this back to the original code.